docs/BUILD.md

```diff
@@ -1,6 +1,36 @@
+
 # Building s2n-tls
 
-s2n-tls can be built as follows:
+To use s2n-tls, you must build the library from the source and then include it in your program.
+
+## Requirements
+
+s2n-tls supports and tests on **x86** and **arm** architectures.
+
+### System requirements
+
+* 20GB RAM available, recommended.
+
+### Software requirements
+
+Building s2n-tls requires:
+
+1. Git
+1. GCC or Clang
+1. CMake
+1. Libcrypto, such as OpenSSL
+1. Platform-specific build tools
```

Comment:
zlib-devel on AL2, as an example, or getting the `correct` ar tooling on others.  This is a placeholder, we'll be able to enumerate these for anything in CI/supported, otherwise, ymmv.